provider/execution/execution.go

```diff
@@ -0,0 +1,371 @@
+	var (
+		clients        []schema.ClientMeta
+		diagsChan      = make(chan diag.Diagnostics)
+		totalResources uint64
+		wg             sync.WaitGroup
+	)
+	clients = e.Table.Multiplex(meta)
+	meta.Logger().Debug("multiplexing client", "count", len(clients), "table", e.Table.Name)
+	defer close(diagsChan)
+	wg.Add(len(clients))
+	for _, client := range clients {
+		go func(c schema.ClientMeta, diags chan<- diag.Diagnostics) {
+			count, resolveDiags := e.callTableResolve(ctx, c, parent)
+			atomic.AddUint64(&totalResources, count)
+			diagsChan <- resolveDiags
+		}(client, diagsChan)
+	}
+	var allDiags diag.Diagnostics
+	go func() {
+		for dd := range diagsChan {
+			allDiags = allDiags.Append(dd)
+			wg.Done()
+		}
+	}()
+	wg.Wait()
+	return totalResources, allDiags
+}
+
+func (e TableExecutor) truncateTable(ctx context.Context, client schema.ClientMeta, parent *schema.Resource) error {
+	if e.Table.DeleteFilter == nil {
+		return nil
+	}
+	if e.Table.AlwaysDelete {
+		// Delete previous fetch
+		client.Logger().Debug("cleaning table previous fetch", "table", e.Table.Name, "always_delete", e.Table.AlwaysDelete)
+		return e.Db.Delete(ctx, e.Table, e.Table.DeleteFilter(client, parent))
+	}
+	client.Logger().Debug("skipping table truncate", "table", e.Table.Name)
+	return nil
+}
+
+// cleanupStaleData cleans resources in table that weren't update in the latest table resolve execution
+func (e TableExecutor) cleanupStaleData(ctx context.Context, client schema.ClientMeta, parent *schema.Resource) error {
+	// Only clean top level tables
+	if parent != nil {
+		return nil
+	}
+	client.Logger().Debug("cleaning table table stale data", "table", e.Table.Name, "last_update", e.executionStart)
+
+	filters := make([]interface{}, 0)
+	for k, v := range e.extraFields {
+		filters = append(filters, k, v)
+	}
+	if e.Table.DeleteFilter != nil {
+		filters = append(filters, e.Table.DeleteFilter(client, parent)...)
+	}
+	return e.Db.RemoveStaleData(ctx, e.Table, e.executionStart, filters)
+}
+
+func (e TableExecutor) callTableResolve(ctx context.Context, client schema.ClientMeta, parent *schema.Resource) (uint64, diag.Diagnostics) {
+	// set up all diagnostics to collect from resolving table
+	var diags diag.Diagnostics
+
+	if e.Table.Resolver == nil {
+		return 0, diags.Append(NewError(diag.ERROR, diag.SCHEMA, e.ResourceName, "table %s missing resolver, make sure table implements the resolver", e.Table.Name))
+
+	}
+	if err := e.truncateTable(ctx, client, parent); err != nil {
+		return 0, diags.Append(FromError(err, WithResource(e.ResourceName), WithErrorClassifier))
+	}
+
+	res := make(chan interface{})
+	var resolverErr error
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				client.Logger().Error("table resolver recovered from panic", "table", e.Table.Name, "stack", string(debug.Stack()))
+				resolverErr = FromError(fmt.Errorf("table resolver panic: %s", r), WithResource(e.ResourceName), WithSeverity(diag.PANIC),
```

Comment:
while we're at it is it possible to think about where would the stacktrace go in resolverErr? maybe a separate error type (and remove diag.PANIC)?

Reply:
It will basically be a copy paste more or less / syntactic sugar so not sure, can add, maybe in a later PR?

Reply:
Adding it as detail was spot on, no need to do anything else I think